### PR TITLE
[Fix] Add method to find the Idp Role Claim URI in expected dialect

### PR DESCRIPTION
**Pull request by inthirakumaaran**
fix wso2/product-is#6396

### CODE REVIEW

**Comment by inthirakumaaran** [comment]
@tharindu-bandara  @JKAUSHALYA

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 1]
```diff
@@ -1782,6 +1782,54 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
...
+
+        // Check whether to use the default dialect or custom dialect
+        if (useDefaultIdpDialect || useLocalClaimDialectForClaimMappings) {
+            String idPStandardDialect = authenticator.getClaimDialectURI();
+            try {
+                // Maps the idps dialect to standard dialect
+                carbonToStandardClaimMapping = ClaimMetadataHandler.getInstance()
+                        .getMappingsMapFromOtherDialectToCarbon(idPStandardDialect,
+                                null, context.getTenantDomain(),
+                                false);
```
Comment (tharindu-bandara) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: We might not need a new line here.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 2]
```diff
@@ -1782,6 +1782,54 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
...
+            } catch (ClaimMetadataException e) {
+                carbonToStandardClaimMapping = new HashMap<>();
+            }
+            // check for role claim uri in the idaps dialect
+            for (Entry<String, String> entry : carbonToStandardClaimMapping.entrySet()) {
```
Comment (tharindu-bandara) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Can we put this within the try block?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 3]
```diff
@@ -1782,6 +1782,54 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
...
+        if (useDefaultIdpDialect || useLocalClaimDialectForClaimMappings) {
+            String idPStandardDialect = authenticator.getClaimDialectURI();
+            try {
+                // Maps the idps dialect to standard dialect
+                carbonToStandardClaimMapping = ClaimMetadataHandler.getInstance()
+                        .getMappingsMapFromOtherDialectToCarbon(idPStandardDialect,
+                                null, context.getTenantDomain(),
+                                false);
+            } catch (ClaimMetadataException e) {
+                carbonToStandardClaimMapping = new HashMap<>();
```
Comment (tharindu-bandara) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Shall we add a debug log here?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 4]
```diff
@@ -1782,6 +1782,54 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
...
+     * Get the mapped URI for the IDP role mapping
+     * @param idpRoleClaimUri pass the IdpClaimUri created in getIdpRoleClaimUri method
+     * @param stepConfig Relevant stepConfig
+     * @param context Relevant authentication context
+     * @return idpRole claim uri in IDPs dialect or Custom dialect
+     */
+    public static String getMappedIdpRoleClaimUri(String idpRoleClaimUri, StepConfig stepConfig,
+                                                  AuthenticationContext context) {
+
+        // Set the incoming idpClaimUri
```
Comment (tharindu-bandara) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: This line is redundant. And also better to add a full stop at the end of the comment. Please check others.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 5]
```diff
@@ -1782,6 +1782,53 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
...
+                // check for role claim uri in the idaps dialect.
+                for (Entry<String, String> entry : carbonToStandardClaimMapping.entrySet()) {
+                    if (idpRoleMappingURI.equalsIgnoreCase(entry.getValue())) {
+                        idpRoleMappingURI = entry.getKey();
+                    }
+                }
+            } catch (ClaimMetadataException e) {
+                log.debug("Error in getting the mapping between idps and standard dialect");
+            }
+
```
Comment (tharindu-bandara) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: redundant new line

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 6]
```diff
@@ -1782,6 +1782,53 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
...
+                        .getMappingsMapFromOtherDialectToCarbon(idPStandardDialect, null,
+                                context.getTenantDomain(), false);
+                // check for role claim uri in the idaps dialect.
+                for (Entry<String, String> entry : carbonToStandardClaimMapping.entrySet()) {
+                    if (idpRoleMappingURI.equalsIgnoreCase(entry.getValue())) {
+                        idpRoleMappingURI = entry.getKey();
+                    }
+                }
+            } catch (ClaimMetadataException e) {
+                log.debug("Error in getting the mapping between idps and standard dialect");
```
Comment (tharindu-bandara) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: Let's say error in getting the mapping. And idpRoleMappingURI itself is returned. better to print that URI value in the logs

**Pull request merged**